### PR TITLE
Enable DocuSign for membership agreements

### LINKED ISSUE #2516: Membership agreements on docusign

### LINKED ISSUE #2575: hasSignedLatestAgreement should be a calculated attribute instead of a method

### LINKED ISSUE #2586: Membership agreement email is sent twice

### PR DESCRIPTION
**Pull request by kberzinch**
Closes #2516
Closes #2575 
Closes #2586

### CODE REVIEW

File: app/Models/User.php [hunk 1]
```diff
@@ -848,8 +849,15 @@ static function (QueryBuilder $query): void {
...
                 }
-            )
-            ->exists();
+            );
+
+        if (true === config('features.docusign-membership-agreement')) {
+            $query->whereHas('envelope', static function (Builder $query): void {
+                $query->where('complete', true);
+            });
+        }
```
Comment (JoshOldenburg) [inline app/Models/User.php:858]: Does this handle users who signed before DocuSign? Or is it deliberate that it'll force you to re-sign?

**Review by JoshOldenburg — COMMENTED** [review]
One quick question. Looks good overall

File: app/Models/User.php (hunk 1 above)
Comment (kberzinch) [reply app/Models/User.php:858]: It deliberately forces re-sign

Comment (JoshOldenburg) [reply app/Models/User.php:858]: Sounds good.

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**